src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildTargetTFMSelector.cs

```diff
@@ -70,15 +70,22 @@ public TargetFrameworkMoniker SelectTFM(IProject project)
             var tfm = new TargetFrameworkMoniker(tfmName);
             foreach (var dep in project.ProjectReferences)
             {
-                if (_tfmComparer.IsCompatible(tfm, dep.TFM))
+                try
                 {
-                    continue;
+                    if (_tfmComparer.IsCompatible(tfm, dep.TFM))
+                    {
+                        continue;
+                    }
+
+                    if (dep.TFM.IsNetCore || dep.TFM.IsNetStandard)
+                    {
+                        tfm = dep.TFM;
+                        _logger.LogDebug("Considering TFM {TFM} for project {Project} based on its dependency on {DepProject}", tfm, project.FilePath, dep.FilePath);
+                    }
                 }
-
-                if (dep.TFM.IsNetCore || dep.TFM.IsNetStandard)
+                catch (UpgradeException)
                 {
-                    tfm = dep.TFM;
-                    _logger.LogDebug("Considering TFM {TFM} for project {Project} based on its dependency on {DepProject}", tfm, project.FilePath, dep.FilePath);
+                    _logger.LogWarning($"Unable to determine TFM for dependency {dep.FilePath}; TFM for {project.FilePath} may not be correct.");
```

Comment:
This seems warranted as a defensive coding practice but I don't see how it relates to the main PR focus of project selection.

Does it relate?

Reply:
After this PR, we allow users to upgrade (some) projects in solutions with projects that are invalid due to missing TFMs. This makes sure that if a user upgrades a project that depends on an invalid one, that we warn them we chose their project's new TFM with incomplete information.